components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -759,14 +759,16 @@ public static API addAPIWithGeneratedSwaggerDefinition(APIDTO apiDto, String oas
             apiToAdd.setAsyncApiDefinition(asyncApiDefinition);
         }
 
+        //adding the api
+        apiProvider.addAPI(apiToAdd);
+
+        apiToAdd.setOrganization(organization);
         if (isAsyncAPI) {
             AsyncApiParser asyncApiParser = new AsyncApiParser();
             String apiDefinition = asyncApiParser.generateAsyncAPIDefinition(apiToAdd);
             apiProvider.saveAsyncApiDefinition(apiToAdd, apiDefinition);
```

Comment:
Let's remove this extra persistence layer call and save the definition from the addAPI() method. similar to what we do for openapi definition. See https://github.com/wso2/carbon-apimgt/blob/master/components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java#L223

Set the async definition to the API obj here and handle it from `apiProvider.addAPI(apiToAdd);`

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/10626